addons/account/models/account_move_line.py

```diff
@@ -1932,6 +1944,34 @@ def _prepare_reconciliation_single_partial(self, debit_values, credit_values, sh
             other_aml_values=debit_values,
         )
 
+        if is_cross_currency_payment:
+            if credit_aml['move_type'] == "entry" and credit_aml['payment_date']:
+                debit_currency_rate_during_payment, credit_currency_rate_during_payment = \
+                    self._get_currency_debit_and_credit_rates(debit_currency, credit_currency, credit_aml['payment_date'])
+            elif debit_aml['move_type'] == "entry" and debit_aml['payment_date']:
+                debit_currency_rate_during_payment, credit_currency_rate_during_payment = \
+                    self._get_currency_debit_and_credit_rates(debit_currency, credit_currency, debit_aml['payment_date'])
+            else:
+                debit_currency_rate_during_payment = debit_available_residual_amounts.get(debit_currency, {}).get('rate') \
+                    or debit_currency['rate']
+                credit_currency_rate_during_payment = credit_available_residual_amounts.get(credit_currency, {}).get('rate') \
+                    or credit_currency['rate']
```

Comment:
I find it super long read for not much, something like this looks better imo.
```suggestion
            debit_currency_rate_during_payment = debit_available_residual_amounts.get(debit_currency, {}).get('rate') \
                or debit_currency['rate']
            credit_currency_rate_during_payment = credit_available_residual_amounts.get(credit_currency, {}).get('rate') \
                or credit_currency['rate']
            payment_date = credit_aml.get('payment_date') or debit_aml.get('payment_date')
            if "entry" in [credit_aml.get('move_type'), debit_aml.get('move_type')] and payment_date:
                debit_currency_rate_during_payment, credit_currency_rate_during_payment = \
                    self._get_currency_debit_and_credit_rates(debit_currency, credit_currency, payment_date)
```
It is also possible to escape the `is_invoice = False` from the next if-else block and put it as a default value which would only be change to True in one instance but that's up to you. Number of lines VS number of assignments (1 out of 3 cases would have two assignments)

Reply:
That definitely makes it much simpler. I tried implementing it, but it misses out in fetching the correct rates in some cases. Hence, I think we should keep it as it is.